include/oneapi/dpl/numeric

```diff
@@ -31,6 +31,7 @@
 // ATTENTION!!! Include the header from the C++ standard library before the oneDPL config header to see whether the _PSTL_VERSION macro is defined
 #include <numeric>
 #include "oneapi/dpl/pstl/onedpl_config.h"
+#include "oneapi/dpl/internal/serial_numeric_impl.h"
```

Comment:
I'd suggest to move it to Line45 under the macro to not be dependent on headers' order.

Reply:
Hi, @capatober 
We will meet conflicts if doing so. If we include serial_numeric_impl.h to Line45, the following error will happen:
In file included from /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/execution:43:
/usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/pstl/glue_numeric_impl.h:43:1: error: declaration conflicts with target of using declaration already in scope
reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last)
^
/usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/pstl/glue_numeric_defs.h:31:1: note: target of using declaration
reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last);
^
/home/jinge/Workspace/IntelRepo/oneDPL/include/oneapi/dpl/numeric:68:20: note: using declaration
using oneapi::dpl::reduce;
                   ^
1 error generated.

Reply:
So, let's avoid conflicts using macros for `using oneapi::dpl::reduce` for oneapi/dpl/numeric.
I think if you don't move it you also have conflicts, but don't see those due to the test

Reply:
Hi, @capatober 
Could you elaborate how to use any macro for "using oneapi::dpl::reduce" to avoid conflicts? If we use some macros here, will any pstl tests be broken?
Thanks very much.

Reply:
Just as you use it for `serial_numeric_impl.h`. I don't expect any failures for Parallel API tests.

Reply:
> Just as you use it for `serial_numeric_impl.h`. I don't expect any failures for Parallel API tests.

Do you mean following:
```c++
namespace std
{
#ifdef _ONEDPL_HAS_NUMERIC_SERIAL_IMPL
using oneapi::dpl::reduce;
#endif
}
```
Thanks very much.

Reply:
@rarutyun, can we separate the discussion about including oneDPL names into `namespace std` from this review, or do you think it needs to be resolved before commit?

Historically, the idea of adding overloads with execution policies into `namespace std` was to improve usability and facilitate migration to C++17, by providing standard API even for previous versions of the standard. I see this patch being in the same spirit.

Reply:
Hi, @rarutyun and @akukanov 
I also suggest to have a separate discussion for namespace, this PR is only for adding serial std::reduce.
Thanks very much.